annotator/src/main/java/org/cbioportal/annotator/internal/GenomeNexusImpl.java

```diff
@@ -0,0 +1,614 @@
+/*
+ * Copyright (c) 2016 Memorial Sloan-Kettering Cancer Center.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY, WITHOUT EVEN THE IMPLIED WARRANTY OF MERCHANTABILITY OR FITNESS
+ * FOR A PARTICULAR PURPOSE. The software and documentation provided hereunder
+ * is on an "as is" basis, and Memorial Sloan-Kettering Cancer Center has no
+ * obligations to provide maintenance, support, updates, enhancements or
+ * modifications. In no event shall Memorial Sloan-Kettering Cancer Center be
+ * liable to any party for direct, indirect, special, incidental or
+ * consequential damages, including lost profits, arising out of the use of this
+ * software and its documentation, even if Memorial Sloan-Kettering Cancer
+ * Center has been advised of the possibility of such damage.
+ */
+
+/*
+ * This file is part of cBioPortal CMO-Pipelines.
+ *
+ * cBioPortal is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+package org.cbioportal.annotator.internal;
+
+import org.cbioportal.models.TranscriptConsequence;
+import org.cbioportal.models.AnnotatedRecord;
+import org.cbioportal.models.GenomeNexusAnnotationResponse;
+import org.cbioportal.models.MutationRecord;
+import org.cbioportal.models.GenomeNexusIsoformOverridesResponse;
+import org.cbioportal.annotator.Annotator;
+
+import java.util.*;
+import org.apache.commons.lang.StringUtils;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.web.client.RestTemplate;
+import org.apache.log4j.Logger;
+
+/**
+ *
+ * @author Zachary Heins
+ * 
+ * Use GenomeNexus to annotate mutation records
+ * 
+ */
+
+@Configuration
+public class GenomeNexusImpl implements Annotator {
+    
+    @Value("${genomenexus.base_url}")
+    private  String baseUrl;
```

Comment:
I think you should get rid of this base URL from here and the application.properties file.  If you should just fully form the URLs in the application.properties file and not do the combining in java code.  Maybe its possible to reference #{genomenexus.base_url} in the application.properties class so you don't have to rewrite the base_url in multiple locations, but I think thats a cleaner solution.